.github/scripts/setup_runners.sh

```diff
@@ -0,0 +1,120 @@
+#!/bin/bash
+set -e
+
+#####################################
+# Variables / Parameters / Settings #
+#####################################
+
+# This script allows users to manually assign parameters
+if [ "$#" -ne 3 ] || [ "$1" = "--help" ] || [ "$1" = "-h" ]
+then
+  echo "Please assign at 3 parameters when running this script"
+  echo "Example: $0 \$ACTION \$EC2_INSTANCE_NAMES(,) \$GITHUB_TOKEN"
+  echo "Example: $0 \"run\" \"odfe-rpm-ism,odfe-rpm-sql\" \"<GitHub PAT>\""
+  exit 1
+fi
+
+SETUP_ACTION=$1
+SETUP_INSTANCE=`echo $2 | sed 's/,/ /g'`
+SETUP_TOKEN=$3
+SETUP_AMI_ID="ami-0397210a1b7e7e74c"
+SETUP_AMI_USER="ec2-user"
+SETUP_INSTANCE_TYPE="m5.xlarge"
+SETUP_INSTANCE_SIZE=20 #GiB
+SETUP_KEYNAME="odfe-release-runner"
+SETUP_SECURITY_GROUP="odfe-release-runner"
+SETUP_IAM_NAME="odfe-release-runner"
+GIT_URL_API="https://api.github.com/repos"
+GIT_URL_BASE="https://github.com"
+GIT_URL_REPO="opendistro-for-elasticsearch/opendistro-build"
+
+echo "###############################################"
+echo "Start Running $0 $1 $2"
+echo "###############################################"
+
+###############################################
+# Run / Start instances and bootstrap runners #
+###############################################
+if [ "$SETUP_ACTION" = "run" ]
```

Comment:
Move this if-else tree into functions, i.e. a "run" function, a "terminate" function etc. That will make the script easier to read and maintain.

Reply:
We can definitely sourcing the script and run functions directly.
Will try something similar later and see how it goes.